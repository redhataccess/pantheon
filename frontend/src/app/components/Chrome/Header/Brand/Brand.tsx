import React, { Component } from 'react';
import { Link } from "react-router-dom";
import logo from '../../../../images/Pantheon2-logo-white.png';

class Brand extends Component {
    public state = {
        linkText: 'Pantheon',
        logoWidth: 220
    };

    public render() {

        return (
            <React.Fragment>
                <Link to={'/search'}>
                <div className="logo"><img src={logo} alt={this.state.linkText} width={this.state.logoWidth} /></div>
                </Link>
            </React.Fragment>
        );
    }
}

export { Brand }