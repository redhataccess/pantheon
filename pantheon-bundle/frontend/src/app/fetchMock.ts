if (!self.fetch) {
    self.fetch = (input: RequestInfo, init?: RequestInit | undefined) => {
        const p = new Promise<Response>((resolve, reject) => {
            // just don't do anything. this exists so that our tests don't try and actually resolve fetches,
            // (which is exactly what isomorphic-fetch tried to do while we were importing it...)
            // and calling nothing here allows us to skip all of our fetch.then() and fetch.catch() code.
        })
        return p
    }
}