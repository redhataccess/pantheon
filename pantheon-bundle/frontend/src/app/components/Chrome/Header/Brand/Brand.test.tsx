import React from 'react'
import { Brand }  from './Brand'
import '@app/fetchMock'

import { shallow } from 'enzyme'

describe('Brand tests', () => {
  test('should render Brand component', () => {
    const view = shallow(<Brand />)
    expect(view).toMatchSnapshot()
  })
})
