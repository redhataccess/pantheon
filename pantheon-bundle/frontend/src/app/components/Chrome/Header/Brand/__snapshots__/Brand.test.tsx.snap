// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Brand tests should render Brand component 1`] = `
<Fragment>
  <Link
    to="/search"
  >
    <div
      className="logo"
    >
      <img
        alt="Pantheon"
        src="test-file-stub"
        width={220}
      />
    </div>
  </Link>
</Fragment>
`;
